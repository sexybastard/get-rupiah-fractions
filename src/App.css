.main {
  background-color: rgb(66,181,73);
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.container {
  background-color: white;
  padding: 20px 20px 20px 20px;
  border-radius: 4px;
  -webkit-box-shadow: 0px 3px 9px 2px rgba(0,0,0,0.55);
  -moz-box-shadow: 0px 3px 9px 2px rgba(0,0,0,0.55);
  box-shadow: 0px 3px 9px 2px rgba(0,0,0,0.55);
  width: 50vw;
  height: 30vh;
  flex-direction: column;
  min-height: 250px;
  min-width: 750px;
  margin-top: -10vh;
}
