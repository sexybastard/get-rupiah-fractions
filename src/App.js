import React, { Component } from 'react';
import Header from './components/Header';
import InputNumber from './components/InputNumber';
import DisplayResult from './components/DisplayResult';
import './App.css'

const headerTitle = 'Get Rupiah Fractions'

class App extends Component {
  constructor(props) {
    super(props);
  
    this.state = {
      result: '',
      validateResponse: {}
    };
  }

  setResult = (result, validateResponse) => {
    this.setState({ result, validateResponse })
  }

  render() {
    const { result, validateResponse } = this.state

    return (
      <div id="main" className="main">
        <div id="container" className="container">
          <Header title={headerTitle} />
          <InputNumber setResult={this.setResult} />
          <DisplayResult result={result} validateResponse={validateResponse} />
        </div>
      </div>
    );
  }
}

export default App;
