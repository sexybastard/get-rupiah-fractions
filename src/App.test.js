import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe("Component: App", () => {
  it('App renders correctly without data', () => {
    shallow(<App />);
  });
});

describe("Component: App should match snapshoot", () => {
  it("Initial case", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
})

describe("Component: App should have the correct state evolution", () => {
  it("Ensure component to have a correct initial state", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.state('result')).toBe('')
    expect(wrapper.state('validateResponse')).toEqual({})
  });
});

describe("Component: App should have the correct className", () => {
  it("Ensure Component to have a correct className", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('#main').prop('className')).toEqual('main')
    expect(wrapper.find('#container').prop('className')).toEqual('container')
  });
});