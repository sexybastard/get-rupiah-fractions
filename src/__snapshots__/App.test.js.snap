// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match snapshoot Initial case 1`] = `
Array [
  <div
    className="main"
    id="main"
  >
    <div
      className="container"
      id="container"
    >
      <Header
        title="Get Rupiah Fractions"
      />
      <InputNumber
        setResult={[Function]}
      />
      <DisplayResult
        result=""
        validateResponse={Object {}}
      />
    </div>
  </div>,
]
`;
